djhtml/__main__.py

```diff
@@ -1,210 +1,152 @@
-            if not args.quiet:
-                print(result, end="")
-            sys.exit(1 if args.check and changed else 0)
+        changed = _verify_changed(source, result)
+        if changed:
+            changed_files += 1
+        else:
+            unchanged_files += 1
 
         # Write output file
-        if changed and args.check:
-            changed_files += 1
+        if filename == "-":
+            if not options.check:
+                print(result, end="")
         elif changed:
-            output_filename = input_file.name if args.in_place else args.output_file
             try:
-                output_file = open(output_filename, "w")
+                output_file = open(filename, "w")
                 output_file.write(result)
                 output_file.close()
-                changed_files += 1
             except Exception as e:
+                changed_files -= 1
                 problematic_files += 1
-                if not args.quiet:
-                    print(f"Error writing {output_filename}: {e}", file=sys.stderr)
+                _error(f"Error writing {filename}: {e}")
                 continue
-            if not args.quiet:
-                print(
-                    f"reindented {output_file.name}",
-                    file=sys.stderr,
-                )
-        else:
-            unchanged_files += 1
+            _info(f"reindented {output_file.name}")
 
     # Print final summary
-    if not args.quiet:
-        s = "s" if changed_files != 1 else ""
-        have = "would have" if args.check else "have" if s else "has"
-        print(
-            f"{changed_files} template{s} {have} been reindented.",
-            file=sys.stderr,
+    s = "s" if changed_files != 1 else ""
+    have = "would have" if options.check else "have" if s else "has"
+    _info(f"{changed_files} template{s} {have} been reindented.")
+    if unchanged_files:
+        s = "s" if unchanged_files != 1 else ""
+        were = "were" if s else "was"
+        _info(f"{unchanged_files} template{s} {were} already perfect!")
+    if problematic_files:
+        s = "s" if problematic_files != 1 else ""
+        _info(
+            f"{problematic_files} template{s} could not be processed due to an error."
         )
-        if unchanged_files:
-            s = "s" if unchanged_files != 1 else ""
-            were = "were" if s else "was"
-            print(
-                f"{unchanged_files} template{s} {were} already perfect!",
-                file=sys.stderr,
-            )
-        if problematic_files:
-            s = "s" if problematic_files != 1 else ""
-            print(
-                f"{problematic_files} template{s} could not be processed due to an"
-                " error.",
-                file=sys.stderr,
-            )
 
-    sys.exit(changed_files if args.check else problematic_files)
+    # Exit with appropriate exit status
+    if problematic_files:
```

Comment:
Does it make sense to have the same error number for "there were problematic files" and for "files would be changed"? Maybe better to have different exit codes for that.

Reply:
Yes, that makes sense. [Black does this](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#command-line-options) when using the `--check` argument:

> Return code 0 means nothing would change. Return code 1 means some files would be reformatted. Return code 123 means there was an internal error.

I tried implementing this, but got stuck when calling `djhtml non-existent-file` because the check would have to be done within the `_generate_filenames()` function which cannot access the `problematic_files` variable.

After that, I forgot about it, so `djhtml non-existent-file` now gives and exit status of `0`, which is definitely incorrect. I'll fix this and implement your suggestion as well!

Reply:
@sjoerdjob please review c19abf and click "Resolve conversation" below if you approve.

Reply:
There's no "Resolve conversation" on my end.

Also, that commit breaks the closing of problematic files.

Reply:
Thanks, also solved.